django/db/models/query.py

```diff
@@ -2225,8 +2226,22 @@ def get_current_to_attr(self, level):
         return to_attr, as_attr
 
     def get_current_queryset(self, level):
-        if self.get_current_prefetch_to(level) == self.prefetch_to:
-            return self.queryset
+        warnings.warn(
+            "Calling get_current_queryset() is deprecated. "
+            "Use get_current_querysets() instead.",
+            RemovedInDjango60Warning,
+        )
```

Comment:
This deprecation should be documented and tested, check out [docs](https://docs.djangoproject.com/en/4.2/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature).